### PR TITLE
solution implemented

### PR DESCRIPTION
**Pull request by dreeeamcatcher**

### CODE REVIEW

File: src/main/java/mate/jdbc/dao/ManufacturerDaoImpl.java [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+import mate.jdbc.util.ConnectionUtil;
+
+public class ManufacturerDaoImpl implements ManufacturerDao {
+    public Manufacturer create(Manufacturer manufacturer) {
+        String insertManufacturer = "INSERT INTO manufacturers(name, country) VALUES(?, ?);";
+        int returnGeneratedKeys = Statement.RETURN_GENERATED_KEYS;
+        try (Connection connection = ConnectionUtil.getConnection();
+                 PreparedStatement createManufacturerStatement =
+                         connection
+                                 .prepareStatement(insertManufacturer, returnGeneratedKeys)) {
```
Comment (andriy-shulepa) [inline src/main/java/mate/jdbc/dao/ManufacturerDaoImpl.java:22]:
```suggestion
        try (Connection connection = ConnectionUtil.getConnection();
                 PreparedStatement createManufacturerStatement =
                         connection
                                 .prepareStatement(insertManufacturer,  Statement.RETURN_GENERATED_KEYS)) {
```

File: src/main/java/mate/jdbc/dao/ManufacturerDaoImpl.java [hunk 2]
```diff
@@ -0,0 +1,121 @@
...
+
+    public boolean delete(Long id) {
+        String deleteManufacturer =
+                "UPDATE manufacturers SET is_deleted = TRUE WHERE id = ?;";
+        try (Connection connection = ConnectionUtil.getConnection();
+                 PreparedStatement getManufacturerStatement =
+                         connection.prepareStatement(deleteManufacturer)) {
+            getManufacturerStatement.setLong(1, id);
+            return getManufacturerStatement.executeUpdate() >= 1;
+
```
Comment (andriy-shulepa) [inline src/main/java/mate/jdbc/dao/ManufacturerDaoImpl.java:99]:
```suggestion
            return getManufacturerStatement.executeUpdate() > 0;
```
it's better to compare with zero

File: src/main/java/mate/jdbc/lib/Dao.java [hunk 3]
```diff
@@ -5,4 +5,5 @@
 
 @Retention(RetentionPolicy.RUNTIME)
 public @interface Dao {
+
```
Comment (andriy-shulepa) [inline src/main/java/mate/jdbc/lib/Dao.java:8]: needless change

**Review by andriy-shulepa — CHANGES_REQUESTED** [review]

File: src/main/java/mate/jdbc/dao/ManufacturerDaoImpl.java [hunk 4]
```diff
@@ -0,0 +1,119 @@
...
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't delete manufacturer by id=" + id, e);
+        }
+
+    }
+
+    private List<Manufacturer> unpackResultSet(ResultSet resultSet) {
+        List<Manufacturer> manufacturersList = new ArrayList<>();
+        try {
```
Comment (Rommelua) [inline src/main/java/mate/jdbc/dao/ManufacturerDaoImpl.java:106]: better add `throws `to the method signature
```suggestion
```

**Review by Rommelua — APPROVED** [review]

**Pull request closed without merging**